addons/website/static/tests/tours/edit_link_popover.js

```diff
@@ -188,18 +188,7 @@ wTourUtils.registerWebsitePreviewTour('edit_link_popover', {
         content: "Ensure that a click on the link popover link opens a new window",
         trigger: 'iframe .o_edit_menu_popover a.o_we_url_link[target="_blank"]',
         extra_trigger: 'iframe .o_edit_menu_popover a.o_we_full_url[target="_blank"]',
-        run: function (actions) {
-            this.$anchor[0].addEventListener('click', (ev) => {
-                const originalPreventDefault = ev.preventDefault.bind(ev);
-                ev.preventDefault = () => {
-                    throw new Error(
-                        "The link popover should not be default prevented as to open a new tab"
-                    );
-                };
-                originalPreventDefault(); // We do not want to open a new tab in a tour
-            }, {once: true, capture: true});
-            actions.click();
-        },
+        run: 'click'
```

Comment:
`click` is the default `run` so in this case you would simply remove it.
But I'd like to have @detrouxdev opinion about this, because he wrote that step on purpose (see https://github.com/odoo/odoo/commit/f9e51a8ab1cc469bd3dd4dba21856597d5e0f902) but I don't understand what it does.

It's supposed to fix a runbot race condition so I don't want to revive it

I saw one of your message on that PR https://github.com/odoo/odoo/pull/114470#issuecomment-1463907491:
> About fixing that however, I guess there's multiple ways of handling it:
>  [...]
> Change the link popover clicks **to not rely on the default behaviour**

It's what is done here.

Reply:
The step is in reference to [this comment](https://github.com/odoo/odoo/pull/114470#pullrequestreview-1333182991), more specifically:

> Could be nice to have a test to enforce this... although not sure how we would detect a link being opened in a new tab

The goal was to find a way to ensure a tab was opened after clicking the link. We settled on assuming that if the click event was not "preventDefault"ed, it would open on a tab.

However here, with this change, this line would not test anything. As it's impossible to check the amount of tabs that are open in the headless chrome browser. (It might be possible Python side but I don't think the test framework is made for that, might be worth checking?).

We might be able to check if the window.open method was properly called though. If instead of directly calling window.open you import browser from "@web/core/browser/browser" and in the test you patch that object to replace the open method with one that checks if the URL is the correct one. You can check `rpc_services_test.js` for an example, though it's done in QUnit, I'm pretty sure we could find a way to do it in the tour.

It might be overkilled but I think it's worth testing these kind of features as we could easily overlook it.

Reply:
Thanks for the explanation :+1: @divy-odoo could you do that please?

Note:

>  We settled on assuming that if the click event was not "preventDefault"ed, it would open on a tab.

Or it could also mean that it was navigating in the current tab, no?

Reply:
Yup, I should have added a step to ensure that we stayed on the same page. That's an oversight 😬